model/testresult/local_service.go

```diff
@@ -266,44 +279,52 @@ func (s *localService) filterTestResults(results []TestResult, opts *FilterOptio
 }
 
 func (s *localService) sortTestResults(results []TestResult, opts *FilterOptions, baseStatusMap map[string]string) {
-	switch opts.SortBy {
-	case SortByStart:
-		sort.SliceStable(results, func(i, j int) bool {
-			if opts.SortOrderDSC {
-				return results[i].TestStartTime.After(results[j].TestStartTime)
-			}
-			return results[i].TestStartTime.Before(results[j].TestStartTime)
-		})
-	case SortByDuration:
-		sort.SliceStable(results, func(i, j int) bool {
-			if opts.SortOrderDSC {
-				return results[i].Duration() > results[j].Duration()
-			}
-			return results[i].Duration() < results[j].Duration()
-		})
-	case SortByTestName:
-		sort.SliceStable(results, func(i, j int) bool {
-			if opts.SortOrderDSC {
-				return results[i].GetDisplayTestName() > results[j].GetDisplayTestName()
-			}
-			return results[i].GetDisplayTestName() < results[j].GetDisplayTestName()
-		})
-	case SortByStatus:
-		sort.SliceStable(results, func(i, j int) bool {
-			if opts.SortOrderDSC {
-				return results[i].Status > results[j].Status
+	sort.SliceStable(results, func(i, j int) bool {
+		for _, sortBy := range opts.Sort {
```

Comment:
is this equivalent to `sort.SliceStable` for each of the sorts one after the other?
like
```go
for _, sortBy := range opts.Sort {
    sort.SliceStable(results, func(i, j int) bool {sortBy.lessThanFunc()}
}
```

Reply:
No because even though it's a stable sort, it will not respect the priority of the sort criteria. For example:
```
[("julian", 5), ("jonathan", 10), ("john", 1), ("april", 9)] 
# sort by (name, int)
# after the first iteration:
[("april", 9), ("john", 1), ("jonathan", 10), ("julian", 5)] 
# and after the second:
[("john", 1), ("julian", 5), ("april", 9), ("jonathan", 10)] 
```

Basically, if we do it one after the other it completely ignores the previous sort criteria. You only want to sort collisions from the previous criteria.

Reply:
ah I understand.
Is it equivalent to doing the `SliceStable`s in reverse, then?

Reply:
Essentially, this attempts to sort by the highest precedent criteria (returning if it's either less than or greater than depending on the sort order). If the keys are equivalent, instead of returning, it continues to the subsequent sort criteria until it can return (either all keys are equal or one is less than the other).

Reply:
Basically, first try sorting by key X, but if the values of key X are equal then try sorting by key Y, and so on.